lib/gym/package_command_generator.rb

```diff
@@ -1,11 +1,37 @@
 require 'shellwords'
 
 module Gym
-  # Responsible for building the fully working xcodebuild command
+  class PackageCommandGenerator
```

Comment:
It's very close in intent :) I have 2 classes and a factory method + a wrapper. That way the client code was not affected. But I can move the factory code to the client or to its own helper method, and expose more the 2 classes.